test/suites/drift/suite_test.go

```diff
@@ -337,6 +337,115 @@ var _ = Describe("Drift", Label("AWS"), func() {
 		Entry("DetailedMonitoring Drift", "DetailedMonitoring", v1alpha1.AWSNodeTemplateSpec{DetailedMonitoring: awssdk.Bool(true)}),
 		Entry("AMIFamily Drift", "AMIFamily", v1alpha1.AWSNodeTemplateSpec{AWS: v1alpha1.AWS{AMIFamily: awssdk.String(v1alpha1.AMIFamilyBottlerocket)}}),
 	)
+	It("should delete a replacement machine from drift if it fails to register", func() {
+		// launch a new machine
+		var numPods int32 = 2
+		dep := test.Deployment(test.DeploymentOptions{
+			Replicas: 2,
+			PodOptions: test.PodOptions{
+				ObjectMeta: metav1.ObjectMeta{Labels: map[string]string{"app": "inflate"}},
+				PodAntiRequirements: []v1.PodAffinityTerm{{
+					TopologyKey: v1.LabelHostname,
+					LabelSelector: &metav1.LabelSelector{
+						MatchLabels: map[string]string{"app": "inflate"},
+					}},
+				},
+			},
+		})
+		env.ExpectCreated(dep, nodeTemplate, provisioner)
+		env.EventuallyExpectNodeCount("==", 2)
+
+		// Drift the machine with bad configuration
+		parameter, err := env.SSMAPI.GetParameter(&ssm.GetParameterInput{
+			Name: awssdk.String("/aws/service/ami-amazon-linux-latest/amzn-ami-hvm-x86_64-ebs"),
+		})
+		Expect(err).ToNot(HaveOccurred())
+		By(fmt.Sprintf("AMI: %s", *parameter.Parameter.Value))
+		nodeTemplate.Spec.AMISelector = map[string]string{"aws::ids": *parameter.Parameter.Value}
+		env.ExpectCreatedOrUpdated(nodeTemplate)
+
+		// Should see the machine has drifted
+		EventuallyWithOffset(1, func(g Gomega) {
+			for _, machine := range env.EventuallyExpectCreatedMachineCount("==", int(numPods)) {
+				g.Expect(env.Client.Get(env, client.ObjectKeyFromObject(machine), machine)).To(Succeed())
+				g.Expect(machine.StatusConditions().GetCondition(v1alpha5.MachineDrifted).IsTrue()).To(BeTrue())
+			}
+		}).Should(Succeed())
+
+		// Expect machine To get cordoned
+		env.EventuallyExpectMachineCount("==", 3)
```

Comment:
Do we need to ensure this again? At this point, we know we have 3 machines (and ideally we should know that we have 3 nodes). We should ensure this check 1 time and then just use the node/machine names to grab an updated version of the data like you do on L380